.github/workflows/release.yaml

```diff
@@ -0,0 +1,247 @@
+
+      - name: Check version
+        id: version_check
+        run: |
+          # Extract version number
+          VERSION=$(awk '/"version":/ { print substr($2, 2, length($2)-3); exit }' app/package.json)
+
+          TAG_VERSION=${{ github.ref_name }}
+          if [ "$VERSION" != "$TAG_VERSION" ]; then
+            echo "Tag version is not correct. Tag version: $TAG_VERSION version from package.json: $VERSION"
+            EC=0
+            exit 1
+          fi
+          echo "VERSION=$VERSION" >> $GITHUB_OUTPUT
+          echo "CONTAINER_TAG=${{ env.ECR_REPO }}:$VERSION" >> $GITHUB_OUTPUT
+      
+      - name: Check if image exist
+        run: |
+          set +e
+          aws ecr describe-images --registry-id "192549843005" --repository-name "concordium/desktop-wallet-ci" --region eu-west-1  --image-ids=imageTag=${{ steps.version_check.outputs.VERSION }}
+          echo "EC=$?" >> $GITHUB_ENV
+
+      - name: build ci image
+        if: ${{ env.EC == 254 }}
+        uses: docker/build-push-action@v6
+        with:
+          push: true
+          tags: "${{ steps.version_check.outputs.CONTAINER_TAG }}"
+          context: '.'
+          file: 'scripts/desktop-wallet-ci.Dockerfile'
+          build-args: |
+            BASE_VERSION=${{ env.BASE_IMAGE_VERSION }}
+            NODE_VERSION=${{ env.NODE_VERSION }}
+
+  release-desktop-wallet-linux:
+    environment: release
+    needs: login-aws
+    runs-on: ubuntu-latest
+    container:
+      image: "${{needs.login-aws.outputs.CONTAINER_TAG}}"
+      credentials:
+        username: ${{needs.login-aws.outputs.access_key}}
+        password: ${{needs.login-aws.outputs.secret_key}}
+      options: -u root
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          ref: ${{ github.ref_name }}
+
+      - name: Install AWS CLI
+        run: apt install awscli -y
+
+      - name: aws creds
+        uses: aws-actions/configure-aws-credentials@v4
+        id: creds
+        with:
+          aws-region: "eu-west-1"
+          role-to-assume: ${{ env.AWS_ROLE_TO_ASSUME }}
+          role-session-name: ReleaseDesktopWalletSession
+          mask-aws-account-id: false
+
+      - name: check if executable exist
+        run: |
+          set +e
+          TARGET_NET=${{env.TARGET_NET}}
+
+          # Prepare filenames
+          if [ $TARGET_NET = "mainnet" ]; then
+             FILENAME="concordium-desktop-wallet-${{ needs.login-aws.outputs.VERSION }}"
+          else
+             FILENAME="concordium-desktop-wallet-${TARGET_NET}-${{ needs.login-aws.outputs.VERSION }}"
+          fi
+
+          FILENAME_LATEST_LINUX="latest-linux.yml"
+
+          OUT_FILENAME="${{ needs.login-aws.outputs.VERSION }}/${TARGET_NET}/${FILENAME}"
+          OUT_LATEST_LINUX="${{ needs.login-aws.outputs.VERSION }}/${TARGET_NET}/latest-linux.yml"
+          aws s3 ls ${{env.S3_BUCKET}}/${OUT_FILENAME}.deb
```

Comment:
Move this into a separate task, and throw an error given any of the files already exists. 

So basically just verify whether the folder already exists